src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/framework/core/RcaConf.java

```diff
@@ -98,6 +100,14 @@ public int getPerVertexBufferLength() {
   }
 
   public HighHeapUsageOldGenRcaConfig getHighHeapUsageOldGenRcaConfig() {
-    return conf.getHighHeapUsageOldGenRcaConfig();
+    return new HighHeapUsageOldGenRcaConfig(conf.getRcaConfigSettings(HighHeapUsageOldGenRcaConfig.CONFIG_NAME));
+  }
+
+  public HighHeapUsageYoungGenRcaConfig getHighHeapUsageYoungGenRcaConfig() {
+    return new HighHeapUsageYoungGenRcaConfig(conf.getRcaConfigSettings(HighHeapUsageYoungGenRcaConfig.CONFIG_NAME));
```

Comment:
I think it would make more sense to just pass the rcaConfig object to the constructor and have it read the config from it. 
Similarly for the `OldGenConfig`

Reply:
done.